plugins/woocommerce/src/Internal/ReceiptRendering/ReceiptRenderingEngine.php

```diff
@@ -197,20 +276,24 @@ private function get_order_data( WC_Order $order ): array {
 					( wc_get_product( $line_item_product->get_parent_id() )->get_name() ) . '. ' . $line_item_product->get_attribute_summary() :
 					$line_item_product->get_name();
```

Comment:
This wasn't touched by this PR, but be aware that `$line_item_product` can be `false`. For example, if the product has been removed since the order was created. There would still be a name to be used in the line item itself, but the product won't exist and this would produce a fatal error.

I believe we should check that ` $line_item_product` is not false before using it.

Reply:
So if `$line_item_product` is false, fallback to simply `$line_item->get_name()`?

Reply:
Yes. Sounds good!

Reply:
I've moved this fix here: https://github.com/woocommerce/woocommerce/pull/47300